packages/ra-core/src/form/useApplyArrayInputDefaultValues.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect } from 'react';
+import {
+    FieldValues,
+    UseFieldArrayReturn,
+    useFormContext,
+} from 'react-hook-form';
+import get from 'lodash/get';
+import { useRecordContext } from '../controller';
+import { InputProps } from './useInput';
+
+type Props = Partial<InputProps> & {
+    source: string;
+};
+
+/*
+ * This hook updates the array input with the default value if default value is present
+ * and field input is not already populated or dirty
+ */
+export const useApplyArrayInputDefaultValues = (
```

Comment:
I felt there is sufficient separation of logic for array inputs to use an new hook over further branching within the existing `useApplyInputDefaultValues` hook.  Happy to refactor into one and use additional props to branch if preferred.